mllib/src/main/scala/org/apache/spark/mllib/tree/model/treeEnsembleModels.scala

```diff
@@ -143,17 +141,9 @@ class GradientBoostedTreesModel(
     val broadcastWeights = sc.broadcast(treeWeights)
 
     (1 until numIterations).map { nTree =>
-      predictionAndError = remappedData.zip(predictionAndError).mapPartitions { iter =>
-        val currentTree = broadcastTrees.value(nTree)
-        val currentTreeWeight = broadcastWeights.value(nTree)
-        iter.map {
-          case (point, (pred, error)) => {
-            val newPred = pred + currentTree.predict(point.features) * currentTreeWeight
-            val newError = loss.computeError(newPred, point.label)
-            (newPred, newError)
-          }
-        }
-      }
+      predictionAndError = GradientBoostedTreesModel.updatePredictionError(
+        remappedData, predictionAndError, broadcastWeights.value(nTree),
```

Comment:
You should pass the broadcast variables themselves; don't extract the value here since this happens before the mapPartitions().